src/OutputHash.js

```diff
@@ -128,23 +128,28 @@ OutputHash.prototype.apply = function apply(compiler) {
             const nonManifestChunks = this.chunks.filter(
                 chunk => !this.manifestFiles.includes(chunk.name)
             );
+
             const chunksByDependency = [];
 
             while (nonManifestChunks.length) {
                 let i = 0;
 
                 while (i < nonManifestChunks.length) {
-                    const current = nonManifestChunks[i];
-
-                    if (
-                        !current.parents
-                        || current.parents.length === 0
-                        || current.parents.every(p =>
-                            chunksByDependency.indexOf(p) !== -1
-                            || nonManifestChunks.indexOf(p) === -1
-                        )
-                    ) {
-                        chunksByDependency.push(current);
+                    const cur = nonManifestChunks[i];
+                    const parents = Array.from(cur.groupsIterable).reduce((acc, group) =>
```

Comment:
webpack@4 allows nested chunk groups, so this won't grab all parents.  I've created a fork with a commit that:
* [Extends this implementation to recurse up the hierarchy](https://github.com/GoodForOneFare/webpack-plugin-hash-output/commit/9677658d9b4c7432780648b3d335cd1fab291f77#diff-a23767514110350738fd8d3d9e133e76R104)
* And [a test that fails without this fix](https://github.com/GoodForOneFare/webpack-plugin-hash-output/commit/9677658d9b4c7432780648b3d335cd1fab291f77#diff-c1129c8b045390789fa8ff62f2c6b4a9R134)

It'd be great if that could be incorporated into this PR!  Otherwise, let me know and I'll create a fresh PR.